test/unit/org/apache/cassandra/index/sai/metrics/QueryMetricsTest.java

```diff
@@ -225,6 +226,11 @@ public void testKDTreeQueryMetricsWithSingleIndex()
     @Test
     public void testKDTreePostingsQueryMetricsWithSingleIndex()
     {
+        // Turn off the query optimizer.
+        // We need to do this in order to remove unpredictability of query plans, so that we get consistent metrics.
+        // We don't want the query optimizer to eliminate the use of indexes.
+        QueryController.QUERY_OPT_LEVEL = 0;
```

Comment:
I was wondering, don't we need to restore it at the end

Reply:
Makes sense

Reply:
Ah, just  noticed this is automatically restored before each test in `SAITester`. So no need to restore.